hist/hist/inc/TF1.h

```diff
@@ -352,15 +352,6 @@ class TF1 : public TNamed, public TAttLine, public TAttFill, public TAttMarker {
       ROOT::Internal::TF1Builder<Func>::Build(this, f);
    }
 
-   // backward compatible interface
-   template <typename Func>
-   TF1(const char *name, Func f, Double_t xmin, Double_t xmax, Int_t npar, const char *, EAddToList addToGlobList = EAddToList::kDefault) :
```

Comment:
Should we deprecate them first or are we sure they were only use by/via CINT?

Reply:
I don't see why anyone would use these constructors otherwise, as it's not needed in C++ to pass the name of the class  instance `Func` as a separate argument.

The reason why I think we can remove it without deprecation is that the replacement is trivial and backwards compatible: just drop these `const char*` inputs at the end. But if you're more comfortable with a deprecation period, I have no problem.

Maybe @lmoneta also has a opinion there?